### PR TITLE
feat(windows) powershell and batch support (fix mac)

### LINKED ISSUE #10: Windows support?
Currently, this is unknown. I'm pretty sure the executor logic will need to be adjusted for Windows.

### PR DESCRIPTION
**Pull request by tensor-programming**
### Which issue does this fix?

Closes #10

### Describe the solution
Adds Batch and powershell support to Mask.

### CODE REVIEW

**Comment by jakedeichert** [comment]
Discussion over here: https://github.com/jakedeichert/mask/pull/45
^ for context if anyone comes and reads this PR

---

@tensor-programming thank you for reopening this PR 👍 

I'll find time to figure out the macOS issue this weekend hopefully. Surprised CI/linux has no issues but mac does 😂

**Comment by tensor-programming** [comment]
What actually going on on Mac?  Ill get one of my guys to look at it.

**Comment by jakedeichert** [comment]
After building this branch and running that version of mask locally with the root maskfile, this error occurs:

```
ERROR: No such file or directory (os error 2)
```

When I removed the powershell code blocks, I believe it fixed the issue. So it might be something to do with mac trying to run the powershell block or something.

I didn't get to it this weekend unfortunately. Might tackle in the next few nights.

**Comment by tensor-programming** [comment]
What I can do is update the CI for mac and see if I can get it to build with my fixes.

**Comment by tensor-programming** [comment]
Interesting, it seems that at least on the Github virtual machine there are no errors (running the tests).  I will have to get one of my guys to check it out since I don't have a mac myself.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,16 +36,16 @@ jobs:
...
         run: cargo test --frozen
 
-  format:
-    name: Format
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout master
-        uses: actions/checkout@master
-      - name: Verify formatting is correct
-        run: cargo fmt --all -- --check
```
Comment (jakedeichert) [inline .github/workflows/ci.yml:43]: I think you accidentally removed formatting from CI? I'll fix on master so i can get this merged.

Comment (tensor-programming) [reply .github/workflows/ci.yml:43]: I guess I did.  Woops, sorry about that.

**Comment by tensor-programming** [comment]
Nice to see that this got merged without issue.  Glad it wasn't some esoteric problem and just a little user error (happens to the best of us).  cheers mate.

**Pull request merged**